src/refinement/refinement.cpp

```diff
@@ -130,6 +130,50 @@ AmrTag FirstDerivative(MeshBlock *pmb, const ParArrayND<Real> &q,
   return AmrTag::same;
 }
 
+AmrTag SecondDerivative(MeshBlock *pmb, const ParArrayND<Real> &q,
+                        const Real refine_criteria, const Real derefine_criteria) {
+  const int dim1 = q.GetDim(1);
+  const int dim2 = q.GetDim(2);
+  const int dim3 = q.GetDim(3);
+  int kl = 0, ku = 0, jl = 0, ju = 0, il = 0, iu = 0;
+  if (dim3 > 1) {
+    kl = 1;
+    ku = dim3 - 2;
+  }
+  if (dim2 > 1) {
+    jl = 1;
+    ju = dim2 - 2;
+  }
+  if (dim1 > 1) {
+    il = 1;
+    iu = dim1 - 2;
+  }
```

Comment:
@pgrete I have no idea.  I don't know how many ghost zones are used in Parthenon or how to know if I'm looking at zone, face, edge, ... centered data.  I simply followed the pattern used in the first derivative refinement criterion assuming that it tested the appropriate domain.

Please feel free to update both order 1 and 2 to use the appropriate domain.

Reply:
The number of ghost zones is an input parameter.
I'll put the question on the agenda for the call tomorrow so that we either fix this or add a comment why this is done.
I'm thinking about downstream development here, where a new developer tries to implement a new criterion and is wondering the same thing.

Reply:
Sounds good - I won't be there tomorrow, so just post the decision or feel free to make the changes.